docs/api/caching.md

```diff
@@ -35,3 +36,252 @@ memory, letting you pick up where you left off.
     `**/__marimo__/cache/` to your `.gitignore`.
 
 ::: marimo.persistent_cache
+
+## Caching expectations
+
+The performance and reliability of your notebook may depend on the caching functionality—as such, it is important to understand the expectations of marimo's caching utilities.
+
+`mo.cache` context blocks are designed to provide deterministic cache restoration when using marimo's APIs for state management.
+
+- On cache hit, the cached values are restored and the code block is not executed, meaning no side effects like printing, file I/O, or network requests will occur.
+- Rerunning a cell will not remove your cache, enabling cache hits on subsequent runs.
+- Dynamic changes to the notebook will appropriately invalidate the cache, if either:
+    1. the code in the defined block changes,
+    2. the variables with basic data types used by your code block change,
+    3. or the relevant code that defines complex datatype variables changes.
+- External modules are assumed to not mutate variables in a way that would affect the cache.
+    - By setting `pinned_version` to `True`, you can ensure that the cache is invalidated when module versions change.
+    - This limitation does not apply if the external module is a marimo notebook.
+- Return values must be serializable.
```

Comment:
these bullets feel like a mix of contract vs expectations. maybe this last could be re-worded or split it up